Jenkinsfile

```diff
@@ -3,767 +3,925 @@
 import groovy.json.JsonOutput
 import bcgov.GitHubHelper
 
+
+// Notify stage status and pass to Jenkins-GitHub library
+void notifyStageStatus (String name, String status) {
+    GitHubHelper.createCommitStatus(
+        this,
+        GitHubHelper.getPullRequestLastCommitId(this),
+        status,
+        "${env.BUILD_URL}",
+        "Stage '${name}'",
+        "stage/${name.toLowerCase()}"
+    )
+}
+
+
+// Create deployment status and pass to Jenkins-GitHub library
+void createDeploymentStatus (String suffix, String status, String targetURL) {
+    def ghDeploymentId = new GitHubHelper().createDeployment(
+        this,
+        "pull/${env.CHANGE_ID}/head",
+        [
+            'environment':"${suffix}",
+            'task':"deploy:pull:${env.CHANGE_ID}"
+        ]
+    )
+
+    new GitHubHelper().createDeploymentStatus(
+        this,
+        ghDeploymentId,
+        "${status}",
+        ['targetUrl':"${targetURL}"]
+    )
+
+    if ('SUCCESS'.equalsIgnoreCase("${status}")) {
+        echo "${suffix} deployment successful!"
+    } else if ('PENDING'.equalsIgnoreCase("${status}")){
+        echo "${suffix} deployment pending."
+    }
+}
+
+
+// Print stack trace of error
+@NonCPS
+private static String stackTraceAsString(Throwable t) {
+    StringWriter sw = new StringWriter();
+    t.printStackTrace(new PrintWriter(sw));
+    return sw.toString()
+}
+
+
+// OpenShift wrapper
+def _openshift(String name, String project, Closure body) {
+    script {
+        openshift.withCluster() {
+            openshift.withProject(project) {
+                echo "Running Stage '${name}'"
+                waitUntil {
+                    notifyStageStatus(name, 'PENDING')
+                    boolean isDone=false
+                    try{
+                        body()
+                        isDone=true
+                        notifyStageStatus(name, 'SUCCESS')
+                        echo "Completed Stage '${name}'"
+                    }catch (error){
+                        notifyStageStatus(name, 'FAILURE')
+                        echo "This is where the errors go!"
+                        echo "${stackTraceAsString(error)}"
+                        def inputAction = input(
+                            message: "This step (${name}) has failed. See error above.",
```

Comment:
This message is shown on in Jenkins as the title to the re-run dialog, which has no context of "above" for errors.